### ISSUE TITLE
Is the gmm reg loss same as total variation loss? 

### ISSUE THREAD

**Issue by Jarvisss** [minar09/cp-vton-plus#31]
I have read the code carefully, the gic loss calculates the difference of flow value at each grid coordinate, on x and y directions relatively.

And the tv loss seems do the same thing. 
https://discuss.pytorch.org/t/implement-total-variation-loss-in-pytorch/55574

Could you please explain the difference between your gic loss and tv loss?

**Comment by thaithanhtuan** [comment]
TV loss and the loss we used are totally different.
Based on the given link, the TV loss is calculated on the pixel value while the loss we used is based on the pixel location.

So, the TV loss is trying to make the smoothness of the warped cloth which is only suitable for non-texture (no logo, no text,...) on the clothes.
 
Our loss is to preserve the affine transform locally. If you increase the hyperparameter of this loss in the total loss, the deformation will be affine transformation locally and globally. But if the hyperparameter is in a reasonable value, the warped-cloth can be deformed with any shape globally and for local, it is affine transformation.

**Comment by Jarvisss** [comment]
Hi, thaithanhtuan, thanks for the quick reply !

The original tv loss is defined on pixel value, as the link does, it tries to regularize the local pixel smoothness.

If we modify the tv loss by changing the pixel value to grid location, then the things are the same, right? 
Just like in [Clothflow](https://openaccess.thecvf.com/content_ICCV_2019/papers/Han_ClothFlow_A_Flow-Based_Model_for_Clothed_Person_Generation_ICCV_2019_paper.pdf), the flow regularization loss. 

![image](https://user-images.githubusercontent.com/16323003/97248964-df199700-183d-11eb-8d64-cf3fd70279e7.png)
